### ISSUE TITLE
Default Hyperparameter Selection Inquiry

### ISSUE THREAD

**Issue by bjh03205** [ultralytics/ultralytics#22837]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello,
First of all, thank you very much for your hard work in developing and maintaining such an excellent model.

While performing hyperparameter tuning, I came across a question and would like to ask for clarification.

Could you explain the criteria and process by which the default optimizer and hyperparameter values were selected for model training?

As I understand it, these default values are generally determined based on configurations that have shown strong performance across various datasets, including COCO. I would like to confirm whether this approach was indeed used to establish the default settings.

If so, I am also wondering whether there are any performance comparison results or benchmark data that the development team used during this process. If available, sharing such information would be extremely helpful.

Thank you very much for your time and assistance.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @bjh03205, thank you for your interest in Ultralytics and for the thoughtful question about default optimizer and hyperparameter choices 🚀 This is an automated response to help you get oriented quickly — an Ultralytics engineer will also review your issue and follow up with more specific guidance soon 🙂

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and background details that may already address parts of your question about training configurations and defaults.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (like your inquiry about how default hyperparameters and optimizers are chosen), please provide as much information as possible about:
-what you are training (task, model, dataset description)  
-any custom hyperparameters or optimizers you are experimenting with  
-relevant training logs, comparisons, or observations  

Also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), as these recommendations closely relate to how defaults are chosen and used in practice ⚙️

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Before diving deeper into optimizer and hyperparameter behavior, please ensure you are on the latest release. Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a **Python>=3.8** environment with **PyTorch>=1.8** to verify everything matches the current defaults and behavior:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for the thoughtful question and for the kind words — a lot of people in the YOLO community have contributed to getting the defaults to where they are today.

At a high level, the default optimizer and hyperparameters come from repeated hyperparameter evolution runs (genetic mutation) on COCO-style detection tasks, starting from YOLOv5 and refined for YOLO11. We search over ranges like those shown in the [Hyperparameter Tuning guide](https://docs.ultralytics.com/guides/hyperparameter-tuning/), then lock in settings that give a good balance of AP, stability, and robustness across different image sizes and batch sizes. The “general” training defaults you see for `lr0`, `momentum`, `weight_decay`, warmup, and loss weights are documented in the [Train settings reference](https://docs.ultralytics.com/modes/train/) and used as the base for most of our benchmarks.

For the optimizer specifically, most users keep `optimizer='auto'`, which is implemented in `BaseTrainer.build_optimizer` and switches between AdamW and SGD and adjusts `lr0` heuristically based on the expected number of iterations and number of classes; you can see the exact logic in the [trainer reference](https://docs.ultralytics.com/reference/engine/trainer/#ultralytics.engine.trainer.BaseTrainer.build_optimizer).

We don’t currently publish a full grid of internal hyperparameter comparison tables, but the COCO metrics reported in the [YOLO11 model overview](https://docs.ultralytics.com/models/yolo11/) are produced with these defaults and serve as the main public benchmark. If you want to mirror our process on your own data, the recommended approach is to start from the documented defaults and then run `model.tune()` with a budget similar to your final training, as described in the [Hyperparameter Tuning guide](https://docs.ultralytics.com/guides/hyperparameter-tuning/).

**Comment by bjh03205** [comment]
Thank you very much for your kind and detailed response.

I have a couple of additional questions for clarification.

When you refer to “COCO-style detection tasks,” does this specifically mean tasks based on the COCO dataset?
In addition, I am not fully certain what it means that the default settings were “ starting from YOLOv5 and refined for YOLO11.”

Based on your explanation, my understanding is that the default hyperparameter values in YOLO models originate from empirically well-performing settings established through collaborative development across versions—from YOLOv5 to YOLO11—using COCO dataset training, and were subsequently refined through genetic mutation–based optimization.
Could you please confirm if this interpretation is correct?

**Comment by glenn-jocher** [comment]
“COCO‑style detection tasks” here means standard object‑detection training on MS COCO and datasets that closely follow its format and statistics (bounding‑box detection, similar object scale/class distributions, COCO‑style labels), not just any arbitrary dataset.

Your summary of how the defaults evolved is essentially correct: we started from empirically strong settings found via large‑scale hyperparameter evolution on COCO with YOLOv5, then carried those forward and re‑tuned them for newer architectures (including YOLO11) using additional evolution runs, ablations, and some manual adjustments; over time this process has been guided by results on COCO and similar benchmarks rather than a single one‑off search, as described more generally in the [hyperparameter tuning overview](https://www.ultralytics.com/glossary/hyperparameter-tuning).
